src/Tizen.NUI/src/public/ParticleSystem/ParticleModifier.cs

```diff
@@ -0,0 +1,167 @@
+    public class ParticleModifierInterface
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ParticleModifierInterface()
+        {
+        }
+
+        /// <summary>
+        /// Second constructor
+        /// </summary>
+        /// <remarks>
+        /// Second constructor should be overriden by the implementation.
+        /// It allows passing variable number of arguments for processing.
+        /// It is called immediately following the class constructor.
+        /// </remarks>
+        /// <param name="list">List of arguments</param>
+        [EditorBrowsable(EditorBrowsableState.Never)] 
+        public virtual void Construct(params object[] list)
+        {
+        }
+
+        /// <summary>
+        /// Updates the ParticleModifier.
+        /// </summary>
+        /// <remarks>
+        /// This callback is responsible for updating particles in the system.
+        /// 
+        /// This callback runs on the Update thread! It should avoid using NUI objects.
+        /// 
+        /// </remarks>
+        /// <param name="emitterProxy">Proxy to the ParticleEmitter object</param>
+        /// <param name="particleList">List of particles to be updated by the modifier</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public virtual void Update(ParticleEmitterProxy emitterProxy, List<Particle> particleList)
+        {
+        }
+
+        /// <summary>
+        /// ParticleEmitter proxy that can be accessed by the user implementation
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ParticleEmitterProxy Emitter;
+    }
+    
+    
+    /// <summary>
+    /// Class represents particle modifier
+    /// </summary>
+    /// <remarks>
+    /// ParticleModifier modifies existing particles in the system.
+    /// Modifiers can be stacked (more than one can be added to the ParticleEmitter).
+    /// Output of one modifier becomes input for next modifier.
+    ///
+    /// Modifier calls into the implementation of <see cref="ParticleModifierInterface"> class.
+    /// 
+    /// </remarks>
+    /// <typeparam name="T">Class of interface that derives from <see cref="ParticleModifierInterface"/></typeparam>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public partial class ParticleModifier<T> : BaseHandle where T : ParticleModifierInterface, new()
+    {
+        // static cache for modifiers (binding between native side and interfaces)
+        static ParticleInterfaceRegister<ParticleModifierInterface> gModifierInterfaceRegister = new ParticleInterfaceRegister<ParticleModifierInterface>();
+        
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected override void Dispose(DisposeTypes type)
+        {
+            if (disposed) return;
```

Comment:
```suggestion
            if (Disposed)
            {
                return;
            }
            if (HasBody())
            {
                gModifierInterfaceRegister.Remove(SwigCPtr.Handle);
            }
```

We love to capture by braces even if it is single line.
And also, (actually a lots of people don't doing now but) We'd better use `Disposed` instead of `disposed`.

[Obsolete("Deprecated in API9, will be removed in API11, Use Disposed")]
Look the 'disposed' comments.

And also, I think we have to remove from register here.

Reply:
Thank you, and I wonder where base.Dispose() should go. Just put it under `if (HasBody())`, would you check if that's okay?